spartanstay/spartanstay/src/main/java/com/spartanstay/spartanstay/controller/ListingsController.java

```diff
@@ -18,7 +18,7 @@ public class ListingsController {
     @GetMapping("/rooms")
     String getRooms(@RequestParam("destination") String destination, @RequestParam("checkIn")
                               @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate checkIn, @RequestParam("checkOut")
-    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate checkOut){
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) LocalDate checkOut, @RequestParam("starRatings") String starRatings){
```

Comment:
Can we make this an optional parameter? I think the syntax is required=false in the call to @RequestParam, but Andrew knows more about this

Reply:
We can make it optional as you mentioned. We'd need to replace "starRatings" with required = false in the @RequestParam call

Reply:
It looks like we can do both, although I'm just now learning that we don't actually need that string for most of these parameters since the names are the same.

https://www.baeldung.com/spring-request-param
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/bind/annotation/RequestParam.html

These tags are ultimately just functions, so we can pass a variety of arguments to them (allegedly).

Reply:
I almost wonder if default values would be the easiest solution to this for us, since it would reduce the amount of conditional logic in the service class.